File: Golang/495. Teemo Attacking.go
func findPoisonedDuration(timeSeries []int, duration int) int {
    length := len(timeSeries)
    
    if length < 2 {
        return length * duration
    }
    
    ans := 0
    i := 0
    j := 1
    for j < length {
        gap := timeSeries[j] - timeSeries[i]
        if gap <= duration {
            ans += gap
        } else {
            ans += duration
        }
        i++
        j++
    }
    return ans + duration
}

File: Python3/495. Teemo Attacking.py
class Solution:
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """
        length = len(timeSeries)
        if length < 2:
            return length * duration
        i, j = 0, 1
        total = 0
        while j < length:
            total += min(timeSeries[j]-timeSeries[i], duration)
            i += 1
            j += 1
        return total + duration
